src/str.rs

```diff
@@ -1035,6 +1039,14 @@ mod test {
                     28,
                 )),
             ),
+            (
+                "1.234567890123456789012345678_9",
+                Ok(Decimal::from_i128_with_scale(12345678901234567890123456789, 28)),
+            ),
+            (
+                "0.1234567890123456789012345678_9",
+                Ok(Decimal::from_i128_with_scale(1234567890123456789012345678, 28)),
```

Comment:
For this test, should we not expect `1234567890123456789012345679` (i.e. rounded up?).